KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/actions/ExtractDocumentWithPasswordAction.java

```diff
@@ -0,0 +1,395 @@
+     * password.
+     *
+     * @param password - the password to try when reading the PDF
+     * @param stream   - the stream containing the PDF to read
+     *
+     * @throws IOException
+     * @throws
+     * org.sleuthkit.autopsy.keywordsearch.actions.ExtractDocumentWithPasswordAction.BadPasswordException
+     */
+    private void decryptPdf(String password, ReadContentInputStream stream) throws IOException, BadPasswordException {
+        PDDocument doc;
+        try {
+            doc = PDDocument.load(stream, password);
+        } catch (IOException ex) {
+            throw new BadPasswordException("Unable to load stream stream for pdf with password", ex);
+        }
+        doc.setAllSecurityToBeRemoved(true);
+        try (OutputStream outStream = getOutputStream()) {
+            doc.save(outStream);
+        }
+        if (doc.isEncrypted()) {
+            throw new BadPasswordException("Unable to process: password for encrypted document is not correct");
+        }
+    }
+
+    /**
+     * Get an input stream for the password protected xml based documents such
+     * as xlsx, docx, and pptx.
+     *
+     * @param password - the password to use for reading the file
+     * @param stream   - the input stream containing the file
+     *
+     * @return stream - the inputStream to read the file from
+     *
+     * @throws IOException
+     * @throws GeneralSecurityException
+     * @throws
+     * org.sleuthkit.autopsy.keywordsearch.actions.ExtractDocumentWithPasswordAction.BadPasswordException
+     */
+    private InputStream getOoxmlInputStream(String password, ReadContentInputStream stream) throws IOException, GeneralSecurityException, BadPasswordException {
+        NPOIFSFileSystem filesystem = new NPOIFSFileSystem(stream);
+        EncryptionInfo info = new EncryptionInfo(filesystem);
+        Decryptor decryptor = Decryptor.getInstance(info);
+        if (!decryptor.verifyPassword(password)) {
+            throw new BadPasswordException("Unable to process: password for encrypted document is not correct");
+        }
+        return decryptor.getDataStream(filesystem);
+    }
+
+    /**
+     * Get an output stream for writing of a the unpassword protected copy of
+     * the file.
+     *
+     * @return the EncodedFileOutputStream which the file will be written to.
+     *
+     * @throws IOException
+     */
+    private EncodedFileOutputStream getOutputStream() throws IOException {
+        if (!decryptedFilePathAbsolute.toFile().exists()) {
+            Files.createDirectories(decryptedFilePathAbsolute);
+        }
+        return new EncodedFileOutputStream(new FileOutputStream(
+                decryptedFilePathAbsolute.toString() + File.separator + abstractFile.getName()),
+                TskData.EncodingType.XOR1);
+    }
+
+    /**
+     * Get a password from the user.
+     *
+     * @param title       the title of dialogue to prompt for a password
```

Comment:
For consistency with the rest of the application and the programming universe as a whole, I'd go with "dialog".